### PR TITLE
forgot_fe

### PR DESCRIPTION
**Pull request by qvestyara**
Create forgot password

### CODE REVIEW

File: src/actions/auth/index.js [hunk 1]
```diff
@@ -45,3 +45,43 @@ export const authVerifyEmail = (hash) => {
...
+    return dispatch => {
+        return authApi.forgotEmail(email).then(text => {
+            dispatch({ type: 'EMAIL_EXIST', payload: text });
+        }, err => {    
+            dispatch({ type: 'EMAIL_DOESNOT_EXIST', payload: err })
+        });
+    }
+}
+
+export const authVerifyPasswordReset = (r) => {
```
Comment (carbonid1) [inline src/actions/auth/index.js]: don't make one-letter variables

File: src/api/auth-api.js [hunk 2]
```diff
@@ -71,6 +71,41 @@ export const authApi = {
...
+                res => {
+                    if (res.data.text) {
+                        resolve(res.data.text);
+                    } else reject({status: 401});
+                }, err => reject(err)
+            )
+        })
+    },
 
+    VerifyPasswordReset: (r) => {
```
Comment (carbonid1) [inline src/api/auth-api.js]: one-letter variable

File: src/modules/change-password/ChangePassword.jsx [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    isAuthorized: state.auth.isAuthorized,
+    errMsg: state.auth.errorMessage,
+    confMsg: state.auth.confMsg
+  }),
+  dispatch => ({
+    authVerifyPasswordReset: (r) => dispatch(authVerifyPasswordReset(r)),
+    authChangePassword: (password, passwordConf, r) => dispatch(authChangePassword(password, passwordConf, r))
+    }
+  )
+)(ChangePassword)
```
Comment (carbonid1) [inline src/modules/change-password/ChangePassword.jsx]: put a semicolon at the end of the statement

File: src/actions/auth/index.js [hunk 4]
```diff
@@ -45,3 +45,43 @@ export const authVerifyEmail = (hash) => {
...
+        }, err => {    
+            dispatch({ type: 'RESET_TOKEN_INCORRECT', payload: err })
+        });
+    }
+}
+
+export const authChangePassword = (password, passwordConf, resetToken) => {
+    return dispatch => {
+        return authApi.ChangePassword(password, passwordConf, resetToken).then(text => {
+            dispatch({ type: 'PASSWORD_CHANGE_SUCESS', payload: text });
```
Comment (Jestermaxrko) [inline src/actions/auth/index.js]: change sucess to success

File: src/api/auth-api.js [hunk 5]
```diff
@@ -71,6 +71,41 @@ export const authApi = {
...
+                        resolve(res.data.text);
+                    } else reject({status: 401});
+                }, err => reject(err)
+            )
+        })
+    },
+
+    ChangePassword: (password, passwordConf, resetToken) =>{
+        return new Promise ((resolve, reject) => {
+            httpServise.post(`${host}:${port}/change`, {password: password, passwordConf: passwordConf, resetToken: resetToken}).then(
```
Comment (Jestermaxrko) [inline src/api/auth-api.js]: dont duplicate key and variable, you can use {password, passwordConf, resetToken}

File: src/modules/change-password/ChangePassword.jsx [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+import {authChangePassword}  from './../../actions/auth';
+
+export class ChangePassword extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      password: '',
+      passwordConf: '',
+      formErrors: {   password: '',  passwordConf: ''},
+      passwordValid: true,
```
Comment (Jestermaxrko) [inline src/modules/change-password/ChangePassword.jsx]: why empty password is valid ?

File: src/modules/change-password/ChangePassword.jsx [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    this.props.authChangePassword(this.state.password, this.state.passwordConf, window.location.search);
+  }
+
+  onChangePassword = (e) => {
+    this.setState( 
+      { 'password': e.target.value }, 
+      () => {  if (this.state.passwordConf)
```
Comment (Jestermaxrko) [inline src/modules/change-password/ChangePassword.jsx]: why you call in set state funcion that has no return  ?

File: src/modules/change-password/ChangePassword.jsx [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+  handleSubmit = (e) => {
+    e.preventDefault();
+    this.props.authChangePassword(this.state.password, this.state.passwordConf, window.location.search);
+  }
+
+  onChangePassword = (e) => {
+    this.setState( 
+      { 'password': e.target.value }, 
+      () => {  if (this.state.passwordConf) 
+        this.validateField('passwordConf', this.state.passwordConf)
```
Comment (Jestermaxrko) [inline src/modules/change-password/ChangePassword.jsx]: why you call in set state funcion that has no return ?

File: src/modules/forgot/Forgot.jsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+  onChange = (e) => {
+    this.setState({ [e.target.name]: e.target.value });
+  }
+
+  validateButtonState = () => {
+    this.setState({ formValid: this.validateForm() });
+  }
+
+  validateForm = () => {
+    return this.state.emailValid;
```
Comment (Jestermaxrko) [inline src/modules/forgot/Forgot.jsx]: you dont need a function to get value from state

**Pull request merged**